### PR TITLE
Support fractional shares in TI screen

### PR DESCRIPTION
**Pull request by falnatsheh**

### CODE REVIEW

File: TradeItIosTicketSDK2/TradeItThemeConfigurator.swift [hunk 1]
```diff
@@ -22,6 +22,21 @@ import UIKit
...
 
+    static func configureNumericToggleInput(toggleInput: TradeItNumericToggleInputCell) {
+        toggleInput.backgroundColor = TradeItSDK.theme.backgroundColor
+        toggleInput.toggleView.backgroundColor = TradeItSDK.theme.interactivePrimaryColor.withAlphaComponent(0.1)
+        toggleInput.quantityTypeButton.tintColor = TradeItSDK.theme.interactivePrimaryColor
+        if let label = toggleInput.textLabel {
+            self.styleLabel(label)
+        }
+        
+        toggleInput.textField.attributedPlaceholder = NSAttributedString(
```
Comment (falnatsheh) [inline TradeItIosTicketSDK2/TradeItThemeConfigurator.swift]: `styleTextField` adds a lot of things that is not needed to style this view

File: TradeItIosTicketSDK2.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -33,8 +33,8 @@
 		52E4AE931E4FA36C006F073A /* URLComponents+TradeIt.swift in Sources */ = {isa = PBXBuildFile; fileRef = 52E4AE911E4FA36C006F073A /* URLComponents+TradeIt.swift */; };
 		750100681F1D0FB200CC9E56 /* TradeItBrokerLogoService.swift in Sources */ = {isa = PBXBuildFile; fileRef = 750100671F1D0FB200CC9E56 /* TradeItBrokerLogoService.swift */; };
 		750100691F1D0FB200CC9E56 /* TradeItBrokerLogoService.swift in Sources */ = {isa = PBXBuildFile; fileRef = 750100671F1D0FB200CC9E56 /* TradeItBrokerLogoService.swift */; };
-		750171A8209B68290094B739 /* YahooNumericToggleInputTableViewCell.xib in Resources */ = {isa = PBXBuildFile; fileRef = 750171A7209B68290094B739 /* YahooNumericToggleInputTableViewCell.xib */; };
-		750171A9209B68290094B739 /* YahooNumericToggleInputTableViewCell.xib in Resources */ = {isa = PBXBuildFile; fileRef = 750171A7209B68290094B739 /* YahooNumericToggleInputTableViewCell.xib */; };
+		750171A8209B68290094B739 /* TradeItNumericToggleInputTableViewCell.xib in Resources */ = {isa = PBXBuildFile; fileRef = 750171A7209B68290094B739 /* TradeItNumericToggleInputTableViewCell.xib */; };
```
Comment (jsom) [inline TradeItIosTicketSDK2.xcodeproj/project.pbxproj]: Did you check all the Yahoo screens still work?

File: TradeItIosTicketSDK2/TicketRow.swift [hunk 3]
```diff
@@ -29,7 +29,7 @@ enum TicketRow {
         case .estimatedCost:
             cellReuseId = .readOnly
         case .quantity, .limitPrice, .stopPrice, .amount:
-            cellReuseId = .numericInput
+            cellReuseId = .numericToggleInput
```
Comment (jsom) [inline TradeItIosTicketSDK2/TicketRow.swift]: Is the old numeric type used anywhere anymore? Can we remove the xib, class and this type?

File: TradeItIosTicketSDK2/TradeItNumericToggleInputTableViewCell.xib [hunk 4]
```diff
@@ -1,13 +1,11 @@
 <?xml version="1.0" encoding="UTF-8"?>
-<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="14109" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
+<document type="com.apple.InterfaceBuilder3.CocoaTouch.XIB" version="3.0" toolsVersion="14269.12" targetRuntime="iOS.CocoaTouch" propertyAccessControl="none" useAutolayout="YES" useTraitCollections="YES" useSafeAreas="YES" colorMatched="YES">
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItNumericToggleInputTableViewCell.xib]: If you're on the beta is this going to mess with anyone else that isn't there yet?

File: TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift [hunk 5]
```diff
@@ -452,54 +459,96 @@ class TradeItTradingTicketViewController: TradeItViewController, UITableViewData
...
 
         cell.selectionStyle = .none
-
-        TradeItThemeConfigurator.configure(view: cell)
+        
+        if let toggle = cell as? TradeItNumericToggleInputCell {
+            TradeItThemeConfigurator.configureNumericToggleInput(toggleInput: toggle)
+        } else {
+            TradeItThemeConfigurator.configure(view: cell)
+        }
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift]: This is a bit eh - is there a way we can annotate this cell in a way that the configurator could read without having to have this switch? In some examples in the theme configurator I've used traits and accessibility settings to do this.

File: TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift [hunk 6]
```diff
@@ -33,6 +33,11 @@ class TradeItTradingTicketViewController: TradeItViewController, UITableViewData
         self.reload(row: .marketPrice, animation: .none)
         self.reload(row: .estimatedCost, animation: .none)
     }
+    private var supportedOrderQuantityTypes: [OrderQuantityType] {
+        get {
+            return self.equityOrderCapabilities?.supportedOrderQuantityTypes(forAction: self.order.action, priceType: self.order.type) ?? []
```
Comment (jsom) [inline TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift]: Let's do a rename on `equityOrderCapabilities` to whatever is in the Yahoo screens... either `instrumentOrderCapabilities` or just `orderCapabilities`. It'll just make it easier when refactoring/collapsing these tickets later.

**Comment by falnatsheh** [comment]
@jsom There seems to be code shared between the TI trading screen and `TradeItFxTradingTicketViewController`. Do we want to upgrade FX screen as well?

File: TradeItIosTicketSDK2/TradeItNumericToggleInputTableViewCell.xib (hunk 4 above)
Comment (falnatsheh) [reply TradeItIosTicketSDK2/TradeItNumericToggleInputTableViewCell.xib]: I checked the file on both versions and they could open it with no issues.

**Comment by falnatsheh** [comment]
@jsom I created a [ticket](https://www.pivotaltracker.com/story/show/158506720) that might be blocker to this (along with the FX testing).

**Comment by jsom** [comment]
@falnatsheh which shared code is there?

File: TradeItIosTicketSDK2/TradeItNumericToggleInputCell.swift [hunk 7]
```diff
@@ -5,7 +5,8 @@ class TradeItNumericToggleInputCell: UITableViewCell {
     @IBOutlet weak var quantityTypeButton: UIButton!
     @IBOutlet weak var disclosureIndicatorWidthConstraint: NSLayoutConstraint!
     @IBOutlet weak var disclosureIndicator: DisclosureIndicator!
-
+    @IBOutlet weak var toggleView: UIView!
```
[the remark this answers is not in the document]
Comment (jsom) [reply TradeItIosTicketSDK2/TradeItNumericToggleInputCell.swift]: @falnatsheh is this needed anymore? Looks unused now so we should remove it

File: TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift (hunk 6 above)
Comment (jsom) [reply TradeItIosTicketSDK2/TradeItTradingTicketViewController.swift]: @falnatsheh also don't forget to update this name pls

**Comment by falnatsheh** [comment]
@jsom I initially thought they both used `YahooNumericInputTableViewCell` but it turns out that's not the case (it uses `TradeItNumericInputCell`) so no need to worry about it.

File: TradeItIosTicketSDK2.xcodeproj/project.pbxproj (hunk 2 above)
Comment (falnatsheh) [reply TradeItIosTicketSDK2.xcodeproj/project.pbxproj]: Not all but most, will go over them and check.

File: TradeItIosTicketSDK2/TicketRow.swift (hunk 3 above)
Comment (jsom) [reply TradeItIosTicketSDK2/TicketRow.swift]: @falnatsheh did this get addressed?

**Comment by falnatsheh** [comment]
@jsom there's the theme issue related to this, I have listed it as a blocker [here](https://www.pivotaltracker.com/n/projects/2147881/stories/158275640). Do you still want this in or should we tackle it before we release it?

**Comment by jsom** [comment]
@falnatsheh nah, I don't think it's a blocker to launch. Thanks!

**Pull request merged**